tests/Functional/OembedServiceTest.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_oembed\Functional;
+
+use Drupal\Core\Url;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\oe_oembed\Traits\OembedTestTrait;
+
+/**
+ * Tests the oEmbed provider responses.
+ */
+class OembedServiceTest extends BrowserTestBase {
+
+  use OembedTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'field',
+    'file',
+    'image',
+    'responsive_image',
+    'breakpoint',
+    'media',
+    'oe_oembed',
+    'user',
+  ];
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->createMedia();
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+  }
+
+  /**
+   * Tests that the oEmbed route returns the correct responses.
+   */
+  public function testOembedRoute(): void {
```

Comment:
This function is very complex to understand, can we split by media type and/or add more comments.

Reply:
It's ok. We can do more cleanup in the followups.

Reply:
Can you create follow-ups for this purpose? Thanks.